# Info 2950 Project Phase II

Original Research Question:
What is the effect of average BMI, alocohol consumption, hepititis B vaccinations, developing vs. developed countries, and percent expenditure on health on life expectancy?


# Cleaning Data

First, we import our data set.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
LED = pd.read_csv("Life Expectancy Data.csv")
LED

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358879   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

Next, we change the column names of our dataset by removing leading or ending spaces, changing them to all lowercase, and replacing any remaining spaces with underscores.

In [2]:
#changing column names
new_colnames = [name.strip() for name in LED.columns]
newer_colnames = [name.lower() for name in new_colnames]
newest_colnames = [name.replace(' ','_') for name in newer_colnames]
LED.columns= newest_colnames
LED.head()

country  year      status  life_expectancy  adult_mortality  \
0  Afghanistan  2015  Developing             65.0            263.0   
1  Afghanistan  2014  Developing             59.9            271.0   
2  Afghanistan  2013  Developing             59.9            268.0   
3  Afghanistan  2012  Developing             59.5            272.0   
4  Afghanistan  2011  Developing             59.2            275.0   

   infant_deaths  alcohol  percentage_expenditure  hepatitis_b  measles  ...  \
0             62     0.01               71.279624         65.0     1154  ...   
1             64     0.01               73.523582         62.0      492  ...   
2             66     0.01               73.219243         64.0      430  ...   
3             69     0.01               78.184215         67.0     2787  ...   
4             71     0.01                7.097109         68.0     3013  ...   

   polio  total_expenditure  diphtheria  hiv/aids         gdp  population  \
0    6.0               8.16        65.0       0.1  584.259210  33736494.0   
1   58.0               8.18        62.0       0.1  612.696514    327582.0   
2   62.0               8.13        64.0       0.1  631.744976  31731688.0   
3   67.0               8.52        67.0       0.1  669.959000   3696958.0   
4   68.0               7.87        68.0       0.1   63.537231   2978599.0   

   thinness__1-19_years  thinness_5-9_years  income_composition_of_resources  \
0                  17.2                17.3                            0.479   
1                  17.5                17.5                            0.476   
2                  17.7                17.7                            0.470   
3                  17.9                18.0                            0.463   
4                  18.2                18.2                            0.454   

   schooling  
0       10.1  
1       10.0  
2        9.9  
3        9.8  
4        9.5  

[5 rows x 22 columns]

In [3]:
LED.columns

Index(['country', 'year', 'status', 'life_expectancy', 'adult_mortality',
       'infant_deaths', 'alcohol', 'percentage_expenditure', 'hepatitis_b',
       'measles', 'bmi', 'under-five_deaths', 'polio', 'total_expenditure',
       'diphtheria', 'hiv/aids', 'gdp', 'population', 'thinness__1-19_years',
       'thinness_5-9_years', 'income_composition_of_resources', 'schooling'],
      dtype='object')

In [4]:
LED= LED.drop(['infant_deaths', 'hepatitis_b', 'measles', 'bmi', 'under-five_deaths', 'polio', 'diphtheria', 'thinness__1-19_years', 'thinness_5-9_years'], axis=1)
LED.head()

country  year      status  life_expectancy  adult_mortality  alcohol  \
0  Afghanistan  2015  Developing             65.0            263.0     0.01   
1  Afghanistan  2014  Developing             59.9            271.0     0.01   
2  Afghanistan  2013  Developing             59.9            268.0     0.01   
3  Afghanistan  2012  Developing             59.5            272.0     0.01   
4  Afghanistan  2011  Developing             59.2            275.0     0.01   

   percentage_expenditure  total_expenditure  hiv/aids         gdp  \
0               71.279624               8.16       0.1  584.259210   
1               73.523582               8.18       0.1  612.696514   
2               73.219243               8.13       0.1  631.744976   
3               78.184215               8.52       0.1  669.959000   
4                7.097109               7.87       0.1   63.537231   

   population  income_composition_of_resources  schooling  
0  33736494.0                            0.479       10.1  
1    327582.0                            0.476       10.0  
2  31731688.0                            0.470        9.9  
3   3696958.0                            0.463        9.8  
4   2978599.0                            0.454        9.5

In [5]:
LED.status= pd.Series(LED.status).map({"Developing":0, "Developed":1})
LED= LED.groupby('country').mean()
LED= LED.drop(["year"], axis=1)
LED.sample(10)

status  life_expectancy  adult_mortality  \
country                                                                
Saint Lucia                    0.0         73.60000         146.2500   
Austria                        1.0         81.48125          65.7500   
Armenia                        0.0         73.40000         117.3750   
Iran (Islamic Republic of)     0.0         73.85625         107.0000   
Ecuador                        0.0         74.72500         132.1250   
Eritrea                        0.0         60.68750         308.6250   
Monaco                         0.0              NaN              NaN   
Georgia                        0.0         73.50625         114.9375   
Mozambique                     0.0         53.39375         226.2500   
Maldives                       0.0         75.53750          83.1875   

                              alcohol  percentage_expenditure  \
country                                                         
Saint Lucia                 11.511333                0.000000   
Austria                     12.236000             4928.439188   
Armenia                      3.702667              131.007419   
Iran (Islamic Republic of)   0.016000                0.000000   
Ecuador                      3.807333              109.406051   
Eritrea                      0.608000                6.436069   
Monaco                       0.010000                0.000000   
Georgia                      5.623333               99.525714   
Mozambique                   1.266000               37.034152   
Maldives                     1.407333              626.499821   

                            total_expenditure  hiv/aids           gdp  \
country                                                                 
Saint Lucia                          6.922667   0.15000           NaN   
Austria                              4.715333   0.10000  33827.476309   
Armenia                              4.861333   0.10000   1999.986318   
Iran (Islamic Republic of)           6.290000   0.10000           NaN   
Ecuador                              5.880000   0.19375   2005.352468   
Eritrea                              3.468667   1.21250    259.395356   
Monaco                               4.300000   0.10000           NaN   
Georgia                              7.330667   0.10000   2158.280869   
Mozambique                           5.972667  11.38750    365.068238   
Maldives                             8.236667   0.10000   3818.296962   

                              population  income_composition_of_resources  \
country                                                                     
Saint Lucia                          NaN                         0.664750   
Austria                     6.474880e+06                         0.862375   
Armenia                     1.015165e+06                         0.698375   
Iran (Islamic Republic of)           NaN                         0.715000   
Ecuador                     8.710592e+06                         0.698875   
Eritrea                     2.071919e+06                         0.255312   
Monaco                               NaN                              NaN   
Georgia                     1.015238e+04                         0.677937   
Mozambique                  1.172975e+07                         0.360813   
Maldives                    3.334250e+03                         0.638625   

                            schooling  
country                                
Saint Lucia                  12.70625  
Austria                      15.38750  
Armenia                      11.78750  
Iran (Islamic Republic of)   12.65625  
Ecuador                      13.01250  
Eritrea                       4.92500  
Monaco                            NaN  
Georgia                      12.67500  
Mozambique                    7.98750  
Maldives                     12.00000

In [7]:
LED.dropna()

status  life_expectancy  adult_mortality   alcohol  \
country                                                           
Afghanistan     0.0         58.19375         269.0625  0.014375   
Albania         0.0         75.15625          45.0625  4.848750   
Algeria         0.0         73.61875         108.1875  0.406667   
Angola          0.0         49.01875         328.5625  5.740667   
Argentina       0.0         75.15625         106.0000  7.966667   
...             ...              ...              ...       ...   
Uruguay         0.0         76.07500         119.9375  6.172667   
Uzbekistan      0.0         68.03125         184.8125  1.608667   
Vanuatu         0.0         71.38750         137.8750  0.806667   
Zambia          0.0         53.90625         354.3125  2.239333   
Zimbabwe        0.0         50.48750         462.3750  4.482000   

             percentage_expenditure  total_expenditure  hiv/aids          gdp  \
country                                                                         
Afghanistan               34.960110           8.252500   0.10000   340.015425   
Albania                  193.259091           5.945625   0.10000  2119.726679   
Algeria                  236.185241           4.604000   0.10000  2847.853392   
Angola                   102.100268           3.919333   2.36875  1975.143045   
Argentina                773.038981           6.912667   0.10000  6998.575103   
...                             ...                ...       ...          ...   
Uruguay                  621.838919           8.750000   0.10000  7192.584875   
Uzbekistan                44.373450           5.638000   0.20625   651.092359   
Vanuatu                  282.325746           3.928667   0.10000  2000.245518   
Zambia                    89.650407           5.824000  11.93125   811.811841   
Zimbabwe                  20.364271           6.158667  23.26250   410.980194   

               population  income_composition_of_resources  schooling  
country                                                                
Afghanistan  9.972260e+06                         0.415375    8.21250  
Albania      6.969116e+05                         0.709875   12.13750  
Algeria      2.164983e+07                         0.694875   12.71250  
Angola       1.014710e+07                         0.458375    8.04375  
Argentina    2.012120e+07                         0.794125   16.50625  
...                   ...                              ...        ...  
Uruguay      2.396771e+06                         0.765625   15.23125  
Uzbekistan   9.036317e+05                         0.603000   11.64375  
Vanuatu      1.230962e+05                         0.367500   10.56875  
Zambia       6.260246e+06                         0.498437   11.21250  
Zimbabwe     8.021343e+06                         0.439125    9.82500  

[142 rows x 11 columns]

In [6]:
LED.to_csv("WHOclean.csv")

# Data Description Questions

**What are the observations (rows) and the attributes (columns)?**

The rows of the data set are countries and the columns of the data set are: country (country name), year (year this data was collected), status (developed or developing country), life_expectancy (average life expectancy in age), adult_mortality (probability of dying between 15 and 60 years per 1000 population), infant_deaths (number of infant deaths per 1000 population), alcohol (alcohol consumption recorded per capita (in litres of pure alcohol)), percent_expenditure (expenditure on health as a percentage of GDP per capita), hepitatis_b (Hepatitis B immunization coverage among 1-year-olds as a percentage), measles (number of reported cases per 1000 population), BMI (average Body Mass Index), under_five_deaths (number of under-five deaths per 1000 population), polio (Polio immunization coverage among 1-year-olds as a percentage), total_expenditure (general government expenditure on health as a percentage of total government expenditure), diphtheria (diphtheria tetanus toxoid and pertussis (DTP3) immunization coverage among 1-year-olds as a percentage), hiv/aids (deaths per 1000 live births (0-4 years)), gdp (Gross Domestic Product per capita (in USD)), population, thinness__1-19_years (prevalence of thinness among children and adolescents for ages 10 to 19 as a percentage), thinness_5-9_years (prevalence of thinness among children for age 5 to 9 as a percentage), income_composition_of_resources (Human Development Index in terms of income composition of resources (index ranging from 0 to 1)), schooling (average number of years of schooling).

**Why was this data set created?**

Both the Global Health Observatory (GHO) data repository under the World Health Organization (WHO) and the United Nations track health statistics as well as many other related factors (economic, social, etc.) for all countries. The creators of this data set combined the data from the repository with that of the UN. They observed that in the past 15 years, there have been significant developments in the health sector specifically in terms of decreasing human mortality rates. They created this data set to compare this decreasing mortality rate to a number of other economic, social, and medical factors to analyze how they contribute.

**Who funded the creation of the data set?**

This project was funded by the Industrial Engineering Department at Georgia Institute of Technology.

**What processes might have influenced what data was observed and recorded and what was not?**

In terms of the WHO collecting data, it is possible that more developed countries have more resources and are thus more likely to have more data. With this said, in order to compare each country equitably, the data collectors must collect the same data across all countries. This means that it is possible that the data collection was limited by what data variables were available across every country.

In terms of the creation of the actual data set, the creators mentioned a few factors they used to limit the data set:
    The creators of this data set noted that there were certain countries originally in this data set that had lots of missing data (they noted lesser known countries like Vanuatu, Tonga, Togo, Cabo Verde etc.). They shared that finding the data for these countries was difficult, so they decided to exclude the countries from the final data set of 193 countries. 
    The creators also noted that “among all categories of health-related factors only those critical factors were chosen which are more representative”. This means that they omitted other factors that they deemed less important or less correlated to the data set.

How they data cleaned: Of all health related categories in the WHO data set, only the critical factors (those with high correlations to mortality rates) were chosen to be included. The creators combined these data sets for the years 2000-2015 for 193 countries using the merge function. To account for the missing values, the developers handled it in R software by using the Missmap command.

**What preprocessing was done, and how did the data come to be in the form that you are using?**

The preprocessing done by the creators of the data set are outlined in two questions above. The preprocessing we did was outlined throughout the jupyter notebook; however, to summarize, we renamed the column names by removing leading or ending spaces, changing them to all lowercase, and replacing any remaining spaces with underscores. We also grouped and averaged all of the data for each country for all the years. 

**If people are involved, were they aware of the data collection and if so, what purpose did they expect the data to be used for?**

The WHO notes that all of their data sets "represent the best estimates using methodologies for specific indicators that aim for comparability across countries and time; they are updated as more recent or revised data become available, or when there are changes to the methodology being used". The people involved in the data collection are aware of their data being collected and understand the WHOs intent to utilize their data to analyze global health trends to serve the greater health of humanity.

**Where can your raw source data be found, if applicable? Provide a link to the raw data (hosted in a Cornell Google Drive or Cornell Box).**

https://drive.google.com/file/d/1uWotsPMAGiU_x2lEgssgJe6DtITk4r1I/view?usp=sharing 

**List specific questions for your peer reviewers and project mentor to answer in giving you feedback on this phase:**

Are there too many or too few variables (columns) being evaluated?
Are our research questions too broad or too narrow?